public/docs/_examples/dependency-injection/dart/web/providers.component.dart

```diff
@@ -0,0 +1,353 @@
+
+const template = "{{log}}";
+
+//////////////////////////////////////////
+@Component (selector: "provider-1", template: "{{log}}", providers:
+//#docregion providers-1
+const [Logger]
+//#enddocregion providers-1
+)
+class ProviderComponent1 {
+  String log;
+
+  ProviderComponent1(Logger logger) {
+    logger.log("Hello from logger provided with Logger class");
+    log = logger.logs[0];
+  }
+}
+//////////////////////////////////////////
+@Component (selector: "provider-2", template: "{{log}}", providers:
+//#docregion providers-2
+const [const Provider(Logger, useClass: Logger)]
+//#enddocregion providers-2
+)
+class ProviderComponent2 {
+  String log;
+
+  ProviderComponent2(Logger logger) {
+    logger.log("Hello from logger provided with Provider class and useClass");
+    log = logger.logs[0];
+  }
+}
+
+//////////////////////////////////////////
+@Component (selector: "provider-3",
+    template: "{{log}}",
+    providers: const [const Provider(Logger, useClass: Logger)]
+/*
+//#docregion providers-3
+    const [provide(Logger, useClass: Logger)]
+//#enddocregion providers-3
+*/
+)
+class ProviderComponent3 {
+  String log;
+
+  ProviderComponent3(Logger logger) {
+    logger.log("Hello from logger provided with useClass");
+    log = logger.logs[0];
+  }
+}
+//////////////////////////////////////////
+class BetterLogger extends Logger {}
+
+@Component (selector: "provider-4", template: "{{log}}", providers:
+//#docregion providers-4
+const [const Provider(Logger, useClass: BetterLogger)]
+//#enddocregion providers-4
+)
+class ProviderComponent4 {
+  String log;
+
+  ProviderComponent4(Logger logger) {
+    logger.log("Hello from logger provided with useClass:BetterLogger");
+    log = logger.logs[0];
+  }
+}
+//////////////////////////////////////////
+
+// #docregion EvenBetterLogger
+@Injectable ()
+class EvenBetterLogger {
+  UserService _userService;
+
+  List <String> logs = [];
+
+  EvenBetterLogger(this._userService);
+
+  log(String message) {
+    message = '''Message to ${ _userService.user.name}: ${ message}.''';
+    print(message);
+    logs.add(message);
+  }
+}
+
+// #enddocregion EvenBetterLogger
+@Component (selector: "provider-5", template: "{{log}}", providers:
+//#docregion providers-5
+const [UserService, const Provider(Logger, useClass: EvenBetterLogger)]
+//#enddocregion providers-5
+)
+class ProviderComponent5 {
+  String log;
+
+  ProviderComponent5(Logger logger) {
+    logger.log("Hello from EvenBetterlogger");
+    log = logger.logs[0];
+  }
+}
+//////////////////////////////////////////
+class NewLogger extends Logger {}
+
+class OldLogger {
+  List <String> logs = [];
+
+  log(String message) {
+    throw new Exception("Should not call the old logger!");
+  }
+}
+
+@Component (selector: "provider-6a", template: "{{log}}", providers:
+//#docregion providers-6a
+const [NewLogger,
+// Not aliased! Creates two instances of `NewLogger`
+const Provider(OldLogger, useClass: NewLogger)]
```

Comment:
not sure. 

Mostly because typescript version works in that way, I don't know what to do. So please the changes you consider necessary.